spec/unit/application/ssl_spec.rb

```diff
@@ -119,55 +116,29 @@ def expects_command_to_fail(message)
     end
 
     it 'detects when a CSR with the same public key has already been submitted' do
-      stub_request(:get, %r{puppet-ca/v1/certificate_request/#{name}}).to_return(status: 404)
       stub_request(:put, %r{puppet-ca/v1/certificate_request/#{name}}).to_return(status: 200)
       stub_request(:get, %r{puppet-ca/v1/certificate/#{name}}).to_return(status: 404)
 
       expects_command_to_pass(%r{Submitted certificate request for '#{name}' to https://.*})
 
-      stub_request(:get, %r{puppet-ca/v1/certificate_request/#{name}}).to_return(status: 200, body: @host[:csr].to_pem)
-      #  we skip :put
       stub_request(:get, %r{puppet-ca/v1/certificate/#{name}}).to_return(status: 404)
 
       expects_command_to_pass
     end
 
-    it "warns if the local CSR doesn't match the local public key, and submits a new CSR" do
-      # write out the local CSR
-      File.open(csr_path, 'w') { |f| f.write(@host[:csr].to_pem) }
-
-      # generate a new host key, whose public key doesn't match
-      private_key = OpenSSL::PKey::RSA.new(512)
-      public_key = private_key.public_key
-      File.open(Puppet[:hostprivkey], 'w') { |f| f.write(private_key.to_pem) }
-      File.open(Puppet[:hostpubkey], 'w') { |f| f.write(public_key.to_pem) }
-
-      # expect CSR to contain the new pub key
-      stub_request(:put, %r{puppet-ca/v1/certificate_request/#{name}}).with do |request|
-        sent_pem = OpenSSL::X509::Request.new(request.body).public_key.to_pem
-        expect(sent_pem).to eq(public_key.to_pem)
-      end.to_return(status: 200)
-      stub_request(:get, %r{puppet-ca/v1/certificate/#{name}}).to_return(status: 404)
-
-      Puppet.stubs(:warning) # ignore unrelated warnings
-      Puppet.expects(:warning).with("The local CSR does not match the agent's public key. Generating a new CSR.")
-      expects_command_to_pass(%r{Submitted certificate request for '#{name}' to https://.*})
-    end
-
     it 'downloads the certificate when autosigning is enabled' do
-      stub_request(:get, %r{puppet-ca/v1/certificate_request/#{name}}).to_return(status: 404)
       stub_request(:put, %r{puppet-ca/v1/certificate_request/#{name}}).to_return(status: 200)
       stub_request(:get, %r{puppet-ca/v1/certificate/#{name}}).to_return(status: 200, body: @host[:cert].to_pem)
 
       expects_command_to_pass(%r{Submitted certificate request for '#{name}' to https://.*})
 
       expect(Puppet::FileSystem).to be_exist(Puppet[:hostcert])
+      expect(Puppet::FileSystem).to_not be_exist(csr_path)
```

Comment:
Do we do anything with the saved CSR?  Do we reuse it or check that it exists during the state machine operation?

Reply:
Nope, it's a write only cache (we write it, but never read it back).